staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -493,6 +507,16 @@ func TestPriorityAndFairnessWithPanicRecoveryAndTimeoutFilter(t *testing.T) {
 
 		reqHandlerCompletedCh, callerRoundTripDoneCh := make(chan struct{}), make(chan struct{})
 		rquestTimesOutPath := "/request/time-out-as-designed"
+
+		go func() {
+			select {
+			case <-stopCh:
+			case <-time.After(caseTimeout):
+				close(reqHandlerCompletedCh)
+				t.Error("time out")
+			}
+		}()
```

Comment:
I see this as a hard stopper, the test must complete within a certain timeout. Can we move this to a function block?
```
func hardStop(t *testing.T, stopCh <-chan struct{}, testTimeout time.Duration, cleanup func()) {
	select {
	case <-stopCh:
		// The test has completed normally.
	case <-time.After(testTimeout):
		cleanup()
		t.Fatalf("the test did not finish within %s", testTimeout)
	}
}
```

we can use it 
```
		go hardStop(t, stopCh, requestTimeout*3, func() {
			close(reqHandlerCompletedCh)
		})
```